integration-cli/push_pull_suite_test.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"os/exec"
+	"strings"
+
+	"github.com/go-check/check"
+)
+
+func init() {
+	check.Suite(newDockerPushPullSuite())
+}
+
+// DockerPushPullSuite provides a isolated daemon that doesn't have all the
+// images that are baked into our 'global' test environment daemon (e.g.,
+// busybox, httpserver, ...).
+//
+// We use it for push/pull tests where we want to start fresh, and measure the
+// relative impact of each individual operation. As part of this suite, all
+// images are removed after each test.
+type DockerPushPullSuite struct {
+	d  *Daemon
+	ds *DockerSuite
+}
```

Comment:
As this is implemented now, the name `DockerPushPullSuite` is misleading, because this isn't usable for pushes (no private registry). I think we should do one of the following:

- This should be renamed to something like `DockerHubPullSuite`, and tests involving a private registry should be part of a different testsuite (possibly `DockerRegistrySuite`, with changes to make it similar to this testsuite).

or

- This testsuite could provide a private registry, and replace `DockerRegistrySuite`.